target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-wxr-5950ax12.dts

```diff
@@ -267,8 +267,9 @@
 &switch {
 	status = "okay";
 
-	switch_lan_bmp = <(ESS_PORT1 | ESS_PORT2 | ESS_PORT3 | ESS_PORT4 | ESS_PORT6)>;
-	switch_wan_bmp = <ESS_PORT5>;
+	switch_lan_bmp = <(ESS_PORT1 | ESS_PORT2 | ESS_PORT3 | ESS_PORT4 | ESS_PORT5)>;
+	switch_wan_bmp = <ESS_PORT6>;
+	malibu_first_phy_addr = <0x1f>;
```

Comment:
This literaly makes no sense.

Can you install mdio-netlink and mdio-tools and post the output of `mdio 9*`:

Reply:
> This literaly makes no sense.

Which part of it doesn't make sense? The `malibu_first_phy_addr`? With the default of 0 all ports are dead :shrug: 

> Can you install mdio-netlink and mdio-tools and post the output of `mdio 9*`:

Can not pull the `kmod-mdio-netlink` package with my current build, rebuilding the image now...

Reply:
It doesnt make sense as first port starts at `0x18` and not `0x1f`, there is nothing according to DTS at `0x1f`

Reply:
> It doesnt make sense as first port starts at 0x18 and not 0x1f, there is nothing according to DTS at 0x1f

Well, I got the `0x1f` from decompiling OEM DTS, and it just works, unlike the default 0. Now compiling with `malibu_first_phy_addr = <0x18>` to check if it would work in the same way...

BTW, the port `0x18` from DTS is not even wired, this device only has 3 QCA8075 1GbE ports wired, first usable port being at `0x19`. The other two ports are individual AQR113c 10GbE, one assigned as lan and another as wan. Here are the mdio outputs:
```
# Only lan1 10GbE or wan ports connected:
root@OpenWrt:~# mdio 9*
 DEV      PHY-ID  LINK
0x00  0x00000000  down
0x08  0x00000000  down
0x18  0x004dd0b1  down
0x19  0x004dd0b1  down
0x1a  0x004dd0b1  down
0x1b  0x004dd0b1  down
0x1c  0x004dd0b1  down
0x1d  0x04820a05  down

# Only lan4 1GbE connected:
root@OpenWrt:~# mdio 9* | grep up
0x19  0x004dd0b1  up

# Only lan3 1GbE connected:
root@OpenWrt:~# mdio 9* | grep up
0x1a  0x004dd0b1  up

# Only lan2 1GbE connected:
root@OpenWrt:~# mdio 9* | grep up
 DEV      PHY-ID  LINK
0x1b  0x004dd0b1  up
```

Side note; the dmesg port link sensing are consistent with the physical ports, no matter what the `malibu_first_phy_addr` is set to (switch-plugging the single cable from wan to lan ports progressively):
```
[   13.311912] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   13.488398] IPv6: ADDRCONF(NETDEV_CHANGE): phy1-ap0: link becomes ready
[   17.200072] nss-dp 3a003000.dp5-syn wan: PHY Link up speed: 1000
[   17.200152] IPv6: ADDRCONF(NETDEV_CHANGE): wan: link becomes ready
[   32.475725] l11: disabling
[  165.917517] nss-dp 3a003000.dp5-syn wan: PHY Link is down
[  175.203318] nss-dp 3a007000.dp6-syn lan1: PHY Link up speed: 1000
[  175.203412] br-lan: port 1(lan1) entered blocking state
[  175.208426] br-lan: port 1(lan1) entered forwarding state
[  266.719659] nss-dp 3a007000.dp6-syn lan1: PHY Link is down
[  266.719944] br-lan: port 1(lan1) entered disabled state
[  271.920602] nss-dp 3a001600.dp4 lan2: PHY Link up speed: 1000
[  271.920684] br-lan: port 2(lan2) entered blocking state
[  271.925342] br-lan: port 2(lan2) entered forwarding state
[  321.840137] nss-dp 3a001600.dp4 lan2: PHY Link is down
[  321.840648] br-lan: port 2(lan2) entered disabled state
[  323.921261] nss-dp 3a001400.dp3 lan3: PHY Link up speed: 1000
[  323.921343] br-lan: port 3(lan3) entered blocking state
[  323.925999] br-lan: port 3(lan3) entered forwarding state
[  402.958730] nss-dp 3a001400.dp3 lan3: PHY Link is down
[  402.959246] br-lan: port 3(lan3) entered disabled state
[  406.080582] nss-dp 3a001200.dp2 lan4: PHY Link up speed: 1000
[  406.080668] br-lan: port 4(lan4) entered blocking state
[  406.085330] br-lan: port 4(lan4) entered forwarding state
```

Reply:
Read the comments in ess dtsi for the explanation.

Reply:
> Read the comments in ess dtsi for the explanation.

I did, but only after I already fixed my issue with `malibu_first_phy_addr = <0x1f>`, which made sense for me at the time, because OAM DTS. And now had I confirmed it works the same with `malibu_first_phy_addr = <0x18>` (`mdio 9*` outpuit identical as above). Earlier on I made a mistake and flashed the image with `malibu_first_phy_addr = <0x23>`, and guess what - it just worked.

So, for this particular device, the default `malibu_first_phy_add = <0>` breaks all the ports, and values >= 0x18 magically make all ports great again?!? Perhaps there is more to the `malibu_first_phy_add` that what the dtsi explanation provides?

Reply:
@Ansuel @robimarko back to the topic, I need this device fixed. Does `malibu_first_phy_addr = <0x18>` make sense then? It _does_ work the same, just checked.

Reply:
Are you sure you did read the thing?
```
		/* This is a special binding that controls how the malibu PHY are
		 * init. This value reflect the PHY addr of the first malibu PHY.
		 * Malibu PHY are in a bundle of 5 PHY.
		 * Some device might have some port not connected.
		 * SSDK still needs the addrs of the first PHY (even if not connected)
		 * to correctly setup the malibu PHY.
		 *
		 * This is needed as previously SSDK based this on the port bmp, but
		 * this can be problematic now that we specify correct bmp.
		 *
		 * Most common configuration have the malibu PHY placed at 0.
		 * But some device might have it placed at address 16.
		 * To drive the correct value, check the port id of the malibu PHY
		 * and try to understand what is the first one in devices where some
		 * port are missing. port_phyinfo is normally the way to go to derive
		 * this value in the few special cases.
		 */
```
it's not a matter of testing random values and see what it works... it a matter of setting the value to the correct address... Also for god sake ignore the OEM DTS as it's full of mistake 99% of the times. If something is confusing from the comment i posted, please tell so we can improve it.

Reply:
I _did_ read the thing, but sorry, this platform is totally new to me, and unlike you guys the SSDK is a voodoo magic to me, I would prefer to stay away from studying that thing in any depth, for the sake of preserving this little mental health I have left... I only wish to fix my device.

> it a matter of setting the value to the correct address...

Setting _what_ to the correct value? A `malibu_first_phy_addr`? And if yes, the 0x18 is a correct value, right?

Reply:
I don't want to be pedantic but it was written.

**This value reflect the PHY addr of the first malibu PHY. Malibu PHY are in a bundle of 5 PHY. Some device might have some port not connected. SSDK still needs the addrs of the first PHY (even if not connected) to correctly setup the malibu PHY.**

And

**To drive the correct value, check the port id of the malibu PHY ... port_phyinfo is normally the way to go to derive this value in the few special cases.**

From your DTSI that again is copied from OEM and still have various useless entry...

```
	qcom,port_phyinfo {
		port@1 {
			port_id = <1>;
			phy_address = <0x18>;
		};
		port@2 {
			port_id = <2>;
			phy_address = <0x19>;
		};
		port@3 {
			port_id = <3>;
			phy_address = <0x1a>;
		};
		port@4 {
			port_id = <4>;
			phy_address = <0x1b>;
		};
		port@5 {
			port_id = <5>;
			ethernet-phy-ieee802.3-c45;
			phy_address = <0x0>;
		};
		port@6 {
			port_id = <6>;
			ethernet-phy-ieee802.3-c45;
			phy_address = <0x8>;
		};
	};
```

port 1 (that is not connected from what you are saying is the first PHY of the bundle and is your addr that is 0x18...

So yes 0x18 is correct. You are still convinced your other address are correct... they are not. Setting the wrong value actually makes the ssdk write fixup to unconnected things and your device is working by luck on default PHY settings with this property set wrong. (again sorry for being a little toxic but there is nothing voodoo here, we tried to give instructions on how to ignore everything and just try to derive the correct value)

Reply:
Thanks for clarifying, makes much more sense now. Changing this PR to 0x18.